controller/appcontroller.go

```diff
@@ -445,6 +456,46 @@ func (ctrl *ApplicationController) Run(ctx context.Context, statusProcessors int
 	<-ctx.Done()
 }
 
+func (ctrl *ApplicationController) updateArgoCDConfigMap() error {
```

Comment:
Instead of inserting overrides into the ConfigMap please can generate overrides on the fly in `GetResourceOverrides` [method](https://github.com/argoproj/argo-cd/blob/master/util/settings/settings.go#L438)